firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -296,15 +303,17 @@ private Void registerAndSaveFid(PersistedInstallationEntry persistedInstallation
               /*fid= */ persistedInstallationEntry.getFirebaseInstallationId(),
               /*projectID= */ firebaseApp.getOptions().getProjectId(),
               /*appId= */ getApplicationId());
-      persistedInstallation.insertOrUpdatePersistedInstallationEntry(
-          PersistedInstallationEntry.builder()
-              .setFirebaseInstallationId(installationResponse.getFid())
-              .setRegistrationStatus(RegistrationStatus.REGISTERED)
-              .setAuthToken(installationResponse.getAuthToken().getToken())
-              .setRefreshToken(installationResponse.getRefreshToken())
-              .setExpiresInSecs(installationResponse.getAuthToken().getTokenExpirationTimestamp())
-              .setTokenCreationEpochInSecs(creationTime)
-              .build());
+      if (installationResponse.getFid() != null) {
```

Comment:
fyi:
If a #createInstallation request gets a successful response, the field "fid" should always be set.

What happens here if the field "fid" is not set? What state will the SDK end up in?

Reply:
This is to differentiate between 200, 5xx & 4xx errors.

200 - createInstallation successful response - fid is not empty
5xx - retry once - empty response & fid
4xx - throws exception

Reply:
If that is what you are doing, then I would recommend your code to express that.
In this case, I would recommend that you check for the 200 HTTP-status instead of the "fid" field.

Reply:
For that: I have to include Response Code as a field in InstallationResponse. Is that what you recommend?

Reply:
since y'all control both sides of this exchange we should be able to use and trust the status codes. imo having canonical status codes as a classification for installation responses is a good idea.

Reply:
Ack. PTAL.